tony-core/src/main/java/com/linkedin/tony/util/Utils.java

```diff
@@ -618,5 +618,50 @@ public static void extractResources() {
     }
   }
 
+  /**
+   * Parses Docker related configs and sets the appropriate container environment variables if Docker is available.
+   * Uses reflection to support older versions of Hadoop.
+   */
+  public static boolean parseDockerConfigs(Configuration tonyConf, Map<String, String> containerEnv, String jobType) {
+    if (tonyConf.getBoolean(TonyConfigurationKeys.DOCKER_ENABLED, TonyConfigurationKeys.DEFAULT_DOCKER_ENABLED)) {
+      String imagePath = tonyConf.get(TonyConfigurationKeys.getContainerDockerKey());
+      if (tonyConf.get(TonyConfigurationKeys.getDockerImageKey(jobType)) != null) {
+        imagePath = tonyConf.get(TonyConfigurationKeys.getDockerImageKey(jobType));
+      }
+      if (imagePath == null) {
+        LOG.error("Docker is enabled but " + TonyConfigurationKeys.getContainerDockerKey() + " is not set.");
+        return false;
+      } else {
+        Class containerRuntimeClass;
+        Class dockerRuntimeClass;
+        try {
+          containerRuntimeClass = Class.forName(Constants.CONTAINER_RUNTIME_CONSTANTS_CLASS);
+          dockerRuntimeClass = Class.forName(Constants.DOCKER_LINUX_CONTAINER_RUNTIME_CLASS);
+        } catch (ClassNotFoundException e) {
+          LOG.error("Docker runtime classes not found in this version ("
+                        + org.apache.hadoop.util.VersionInfo.getVersion() + ") of Hadoop.", e);
+          return false;
+        }
+        if (dockerRuntimeClass != null) {
+          try {
+            String envContainerType = (String) containerRuntimeClass.getField(Constants.ENV_CONTAINER_TYPE).get(null);
+            String envDockerImage = (String) dockerRuntimeClass.getField(Constants.ENV_DOCKER_CONTAINER_IMAGE).get(null);
+            containerEnv.put(envContainerType, "docker");
+            containerEnv.put(envDockerImage, imagePath);
+          } catch (NoSuchFieldException e) {
+            LOG.error("Field " + Constants.ENV_CONTAINER_TYPE + " or " + Constants.ENV_DOCKER_CONTAINER_IMAGE + " not "
+                          + "found in " + containerRuntimeClass.getName());
```

Comment:
Include `e`

Reply:
that is too much stack trace. I think our error info is enough